notes/shelley/CoreDIF.md

```diff
@@ -0,0 +1,153 @@
+
+2. Bearer address obfuscation / protection so as to mitigate attacks.
+
+### Membership in core DIF
+
+- Must be a threshold for joining the network. Can't have all stakeholders.
+- Perhaps: you can only join if you are registered (on chain) as a delegate
+  and you have stake.
+  - I'm willing to be called (I opted into being a delegate).
+  - I am called (I am in top (say) 80% of stakeholders, or the top 100
+    stakeholders).
+    The plan seems to be to limit the actual number of members rather than
+    including top n% stakeholders, which could be arbitrarily-many.
+- Expectation: core nodes don't hold coins, but have stake delegated to them.
+  - Example: IOHK and partners would make a new address with no coins and
+    delegate all their stake to these.
+  - I guess the motivation is to protect the keys that control the coins; the
+    key that controls the delegate isn't so precious, since delegation can be
+    revoked.
+- To take delegation, you don't have to be on the public delegate list.
+- Currently there is no freedom from being delegated to you; you can't
+  deny delegation.
+- There are points at which a holder *can* join and *must* join.
+  Work this out with Lars...
+
+### Routing; building spanning tree(s)
+
+- Publish a POA address (signed) to some others (encrypted with their
+  public keys). On the blockchain? Doesn't have to be the core node's IP
+  address; could be perhaps a bridge relay which will relay back to the core
+  node.
+- Valency min and mix. Perhaps 3 and 10.
+- Each node should randomly choose 3 to 10 others. Perhaps XOR metric of
+  their public keys.
+  - Partitions possible. Would it be a disaster?
+    Detectable? Yes, because you know the whole graph (just not the bearer
+    addresses).
+  - Perhaps recompute the graph by using a different salt for the hash on
+    public keys for XOR metric, until the graph is connected.
+  - Nodes can also publish extra routing that they choose themselves?
+    Addresse of relay nodes?
+  - Need an amount of time between knowing the stake distr of the next epoch
+    and before the network has to be established, in which we can do this
+    network computation.
+    TODO look into this. How long do we have?
+
+- Started talking about QoS/Delta-Q in the core DIF
+  Apparently we're going to do exactly what we do in the public net:
+  choose a bunch of peers based on some XOR, then gather some data about their
+  delta-Q, then choose a subset according to percentiles of delta-Q/performance.
+  Difference here is that the computation of delta-Q is wholistic: we're
+  concerened with the entire path (which can be known here) rather than in the
+  non-core-DIF where we're concerned only with the first hop.
+
+- Some discussion about time constraints RE: ensuring that the slot leader's
+  block goes out earlier enough that it reaches the next slot leader before
+  it's too late.
+  Always going for worst case actually lowers best case throughput (i.e. tps).
```

Comment:
I think we concluded we should send out the block at the start of the slot, and not wait.